Issue3165  Term Entry: setLenient() method (#3234)
Codecademy/docs: content/java/concepts/calendar/terms/setLenient/setLenient.md

---
Title: '.setLenient()' 
Description: 'Sets whether the date/time interpretation should be lenient or not.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Date'
  - 'Methods'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The `**.setLenient()**` method of the `calender` class is used to set the lineacy of the interpretation of date and time. i.e., it is used to specify whether the internpretation of date and time should be lenient or not. 

Reviewer: ```suggestion
The **`.setLenient()`** method of the `Calendar` class is used to set the leniency of the interpretation of date and time. In other words, it is used to specify whether the interpretation should be strict or not. 
```

> The method takes one boolean value as a parameter.

Reviewer: ```suggestion
```

> **Note:** When a `Calendar` is lenient, it allows date and time values that may not make sense in a real-world context. For example, `February 30th` or a time value of `25:70:90` would be accepted. When a `Calendar` is non-lenient, it enforces stricter rules, and attempting to set invalid date or time values will throw exceptions, such as `IllegalArgumentException`.

## Syntax

```pseudo

> calender.setLenient(boolean lenient)

Reviewer: ```suggestion
calender.setLenient(bool_flag)
```

```

> Parameter: set `lenient` as false for more strict interpretation, that is, leniency is turned off. Set true for leniency to be turned on.

Reviewer: ```suggestion

This method has only one parameter, `bool_flag`, which should be set as `false` for a strict interpretation, that is, leniency is turned off. And set to `true` for leniency to be turned on.
```

> > **Note:** You can use the `.isLenient()` method to check if given calender instance is lenient or not.

Reviewer: ```suggestion
> **Note:** The `.isLenient()` method can be used to check if a given calendar is lenient or not.
```

## Example 1

> This example shows what happens when you dont use .setLeient() method.

Reviewer: ```suggestion
This example demonstrates a use case for the  `.setLenient()` method.
```
